### DISCUSSION TITLE
Question about networking on 1.20.5

### DISCUSSION THREAD

**Discussion by MATRIX-feather** [FabricMC/fabric-api#3739] [Mod Dev Support]
### Description

So I've been porting my mod to 1.20.5 recently, it's a mod that acts as a client integration for another Bukkit plugin, and when it comes to networking, I get these errors:
![Screenshot 1](https://github.com/FabricMC/fabric/assets/55654482/3cc96bbb-7840-441c-8811-3ab0b4bdd9a4)
![Screenshot 2](https://github.com/FabricMC/fabric/assets/55654482/29355fc2-be1e-41f6-8867-abc2204b861f)

For the code on the fabric side:
Payload type register: https://github.com/XiaMoZhiShi/FeatherMorphClient/blob/1.20.5/src/main/java/xiamomc/morph/client/ServerHandler.java#L165
Payload definition: https://github.com/XiaMoZhiShi/FeatherMorphClient/blob/1.20.5/src/main/java/xiamomc/morph/client/network/payload/MorphInitChannelPayload.java

And on the server side, I send custom messages by calling the Player#sendPluginMessage() method:
https://github.com/XiaMoZhiShi/FeatherMorph/blob/1.20.5/src/main/java/xiamomc/morph/network/server/MorphClientHandler.java#L75

Somehow the client always sends messages with one byte extra, and when the server sends the message to the client, the client immediately disconnects with an error.

Is it a bug about Fabric API or I did something wrong?

**Comment by davidromrell** [comment]
probably a bit of both, the issue is likely that you are trying to ///@Override///
something that cannot be, unless you are an incredible coder .java files are essentially a powerful config file.

**Reply by MATRIX-feather** [reply]
I managed to fix the problem with this commit: https://github.com/XiaMoZhiShi/FeatherMorphClient/commit/539a71ae9b9026e9660c7e61bb35d5b75c14de25

It seems that the original decoder I use doesn't clear the buffer and somehow makes the game panic, after clearing the buffer everything goes well.

And I also unified the string encode/decode method between the server and the client, now the extra byte has no longer been a problem for us.
